tpu-client/src/nonblocking/tpu_client.rs

```diff
@@ -803,6 +822,14 @@ impl LeaderTpuService {
         self.recent_slots.estimated_current_slot()
     }
 
+    pub fn unique_leader_tpu_sockets(&self, fanout_slots: u64) -> Vec<SocketAddr> {
+        let current_slot = self.recent_slots.estimated_current_slot();
+        self.leader_tpu_cache
+            .read()
+            .unwrap()
+            .get_unique_leader_sockets(current_slot, fanout_slots)
+    }
+
     pub fn leader_tpu_sockets(&self, fanout_slots: u64) -> Vec<SocketAddr> {
```

Comment:
the functionality of this method has now changed (returns non unique leader discovery). Are the methods that call this expecting non unique leader discovery? Concerned that this may break previous expected behavior

Reply:
I renamed the old method to `unique_leader_tpu_sockets` and `get_unique_leader_sockets`, so the idea that nothing has changed and these new methods are not used by anyone in the scope of this PR.

Reply:
ahh got it. not sure how i missed that.

Reply:
ya actually that is a good idea